src/utils/token-standard/v2/settlement-factory.ts

```
@@ -0,0 +1,517 @@
+    ),
+    disclosedContracts: disclosedContracts.map(parseDisclosedContract),
+  };
+}
+
+export async function prepareTokenStandardV2SettlementCommand(
+  params: PrepareTokenStandardV2SettlementCommandParams
+): Promise<PreparedTokenStandardV2SettlementCommand> {
+  if (!isRecord(params.scan) || typeof params.scan.getSettlementFactoryFromRegistry !== 'function') {
+    inputInvalid('scan must provide getSettlementFactoryFromRegistry.', { field: 'scan' });
+  }
+  if (Object.prototype.hasOwnProperty.call(params, 'extraArgs')) {
```

ESLint

[error] Use Object.hasOwn() instead of hasOwnProperty

(e18e/prefer-object-has-own)